src/python/pants/engine/engine_aware.py

```diff
@@ -93,9 +93,9 @@ class SideEffecting(ABC):
     """Marks a class as providing side-effecting APIs, which are handled specially in @rules.
 
     Implementers of SideEffecting classes should ensure that `def side_effected` is called before
-    the class causes side-effects.
+    the class causes side effects.
```

Comment:
I'm curious, what's the rule for what makes it "side effects" and not "side-effects" but it is "side-effecting"? (this is an area I don't have any good feeling for how it works.. as it's quite different from Swedish that simply concatenates the words, so it would be "sideeffecting" in that case.)

Reply:
`side-effecting` is acting as an adjective instead of a noun, so it gets hyphenated even though 'side effect', as a noun, does not.

https://english.stackexchange.com/a/34766